### PR TITLE
devops: fix conda release pipeline (linux-arm64)

### PR DESCRIPTION
**Pull request by mxschmitt**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -24,7 +25,7 @@ jobs:
         # Required for conda-incubator/setup-miniconda@v3
         shell: bash -el {0}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
```
Comment (mxschmitt) [inline .github/workflows/publish.yml:28]: drive-by

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ on:
 jobs:
   deploy-conda:
     strategy:
+      fail-fast: false
```
Comment (mxschmitt) [inline .github/workflows/publish.yml:9]: helps us to identify if only one job in the matrix fails or all.

**Review by dgozman — APPROVED** [review]
